packages/web3-core/src/web3_context.ts

```diff
@@ -167,6 +167,10 @@ export class Web3Context<
 		return this._wallet;
 	}
 
+	public set wallet(wallet: Web3BaseWallet<Web3BaseWalletAccount> | undefined) {
+		this._wallet = wallet;
```

Comment:
If there is already accounts in existing wallet this set will over write? so iterate on `wallet` param and individually set accounts into existing wallet, if wallet already exist.